core/provider/digitalocean/client.go

```diff
@@ -0,0 +1,86 @@
+package digitalocean
+
+import (
+	"context"
+
+	"github.com/digitalocean/godo"
+)
+
+// DoClient defines the interface for DigitalOcean API operations used by the provider
+type DoClient interface {
```

Comment:
if we already implement the interface here, could you move and standardize response code checks to here instead of the callers?